### DISCUSSION TITLE
Should I be interested in this integer problem?

### DISCUSSION THREAD

**Discussion by jjhforrest** [coin-or/Cbc#558] [Discussion]
This is not a very serious discussion, but it would be very interesting if someone could solve the problem quickly.

I look at Cbc's preprocessing and try and see what I can do to improve it.  I have been looking at various ideas which include altering the model by scaling and some simple modifications.  Sometimes I use a miplib example and fix a few variables to make it easier.  Some months ago I played around with the miplib problem mas76 and ended up with a problem of 60 0-1 variables and 7 rows (and 100% dense!).  This was meant to be easier to solve than the original!!

I attach the problem
[mas76mod.mps.txt](https://github.com/coin-or/Cbc/files/9862725/mas76mod.mps.txt)

However this problem does not seem easy to solve.  The continuous relaxation is -10512.  I have given it to a good commercial solver and it has got a solution of -10505, but I am not sure it will finish very soon.  It has been running for two days and done 1.2e9 nodes!  More worryingly, it is slowing down.

Anyway - do people think I should offer it to the miplib test set or should I just think that anyone can invent a difficult to solve problem?

**Comment by Marqu3tte** [comment]
Fscip from Scip suite 8.0.4 found -10507 in a few seconds with 11 threads. Or is this already solved?

$ cat mas76mod.sol
[ Final Solution ]
objective value:                               -10507
C0000002                                            1   (obj:-417)
C0000007                                            1   (obj:-336)
C0000008                                            1   (obj:-337)
C0000009                                            1   (obj:-298)
C0000010                                            1   (obj:-323)
C0000011                                            1   (obj:-266)
C0000012                                            1   (obj:-449)
C0000013                                            1   (obj:-341)
C0000014                                            1   (obj:-423)
C0000017                                            1   (obj:-307)
C0000021                                            1   (obj:-439)
C0000023                                            1   (obj:-513)
C0000024                                            1   (obj:-337)
C0000026                                            1   (obj:-422)
C0000028                                            1   (obj:-365)
C0000031                                            1   (obj:-322)
C0000033                                            1   (obj:-305)
C0000037                                            1   (obj:-337)
C0000039                                            1   (obj:-283)
C0000040                                            1   (obj:-325)
C0000041                                            1   (obj:-329)
C0000042                                            1   (obj:-390)
C0000045                                            1   (obj:-363)
C0000046                                            1   (obj:-412)
C0000049                                            1   (obj:-382)
C0000052                                            1   (obj:-394)
C0000056                                            1   (obj:-377)
C0000057                                            1   (obj:-436)
C0000059                                            1   (obj:-279)

**Reply by jjhforrest** [reply]
Impressive.  Did it show any signs of finishing soon?

**Comment by Marqu3tte** [comment]
The job died after ~2 hours, I think it was out of memory. I have 40 GB mem. I will try again with less threads and such ..

**Comment by Marqu3tte** [comment]
Using that -10507 solution as a mipstart, Neos Gurobi found -10508

 263136355 2464513 -10512.000   59    7 -10507.000 -10512.000  0.05%   3.0 4240s
 263468517 2468581     cutoff   59      -10507.000 -10512.000  0.05%   3.0 4245s
 263792781 2474364 -10512.000   55    7 -10507.000 -10512.000  0.05%   3.0 4250s
H264004969 2159441                    -10508.00000 -10512.000  0.04%   3.0 4253s
 264061303 2159894 -10512.000   59    7 -10508.000 -10512.000  0.04%   3.0 4255s
 264354991 2162396     cutoff   58      -10508.000 -10512.000  0.04%   3.0 4260s
 264654562 2164543     cutoff   60      -10508.000 -10512.000  0.04%   3.0 4265s

**Comment by jjhforrest** [comment]
Playing around I have managed to get -10510 solution - taking billions of nodes.  I think two things are of interest.

1) Equivalent solutions can be found by minimizing the sum of the slacks (at least for objective values -10510 to -10512).  If the sum is >=1 solution can be no better than -10510.
2) Giving Cplex the -10510 solution, I thought it might solve it but I killed it after 18 hours as it looks as if it is slowing down.  After an hour it was doing 85000 nodes a second but towards the end this was down to 650 nodes a second!
